includes/profile.inc

```diff
@@ -47,3 +47,267 @@ function cm_tools_module_disable($modules) {
+ * accepts the sandbox that is passed to the update hook as its first parameter.
+ * The results of the query, and the state of the iterations are stored in the
+ * sandbox.
+ *
+ * @param $sandbox
+ *   The sandbox passed to the drupal hook hook that this function is called from.
+ * @param string $entity_type
+ *   The entity type to iterate over.
+ * @param NULL|string $bundle
+ *   The bundle to iterate over, if NULL then all bundles of the given entity type will be iterated.
+ * @param int $entities_per_pass
+ *   The number of entities to load and process in each batch iteration.
+ *
+ * @return
+ *   An array of entities to process in this iteration.
+ */
+function cm_tools_update_entity(&$sandbox, $entity_type, $bundle = NULL, $entities_per_pass = 10) {
+  $entities = array();
+
+  $callback = function($entity) use (&$entities) {
+    $entities[] = $entity;
+  };
+
+  cm_tools_update_entity_walk($sandbox, $entity_type, $callback, $bundle, $entities_per_pass);
+
+  return $entities;
+}
+
+/**
+ * Iterate over all nodes of a specific type.
+ *
+ * This function is designed to work in the context of a Drupal update hook, it
+ * accepts the sandbox that is passed to the update hook as its first parameter.
+ * The results of the query, and the state of the iterations are stored in the
+ * sandbox.
+ *
+ * @param $sandbox
+ *   The sandbox passed to the drupal hook hook that this function is called from.
+ * @param NULL|string $node_type
+ *   The node type to iterate over, if NULL then all nodes will be iterated.
+ * @param int $nodes_per_pass
+ *   The number of nodes to load and process in each batch iteration.
+ *
+ * @return
+ *   An array of nodes to process in this iteration.
+ */
+function cm_tools_update_node(&$sandbox, $node_type = NULL, $nodes_per_pass = 10) {
+  return cm_tools_update_entity($sandbox, 'node', $node_type, $nodes_per_pass);
+}
+
+/**
+ * Fill in missing default field values on entities.
+ *
+ * When adding a new field to an existing entity type, instances of those
+ * entities will be missing values for that field. Drupal does not ensure that
+ * when loaded they get the default, and they won't be able to be queried etc.
+ *
+ * Entities processed by this function will have the default value set
+ * according to the field configuration, so will match the defaults of newly
+ * created entities.
+ *
+ * This function is designed to work in the context of a Drupal update hook, it
+ * accepts the sandbox that is passed to the update hook as its first parameter.
+ * The results of the query, and the state of the iterations are stored in the
+ * sandbox.
+ *
+ * @param $sandbox
+ *   The sandbox passed to the drupal hook hook that this function is called from.
+ * @param string $entity_type
+ *   The entity type to iterate over.
+ * @param NULL|string $bundle
+ *   The bundle to iterate over or NULL to iterate over all bundles for the
+ *   given entity type.
+ * @param $fields
```

Comment:
It might be nice to wrap this jobber with 2 higher level utilities, both of which could simplify `$fields` to be either a field name or simply array of field names:

- One for where I don't care about languages (equivalent to passing NULL as the value)
- One for where I do care about languages, but would rather not specify which. So `$fields` would similarly be a field name or array of field names, but the default value is populated across all languages.

Reply:
Oh, actually that last bit might be tricky, working out what translations are valid for the entity etc. and then populating the languages as appropriate.

How about, we add a higher level helper, that accepts either a flat array of field names, or a string field name and just does the default language for all those?

Reply:
Ah, OK. Sure sure, that sounds like a plan.

Reply:
Given that entity_translation is basically the only thing that we would use to deal with field translations within an entity, we _could_ do something like this to get which languages should be populated, I believe: 

```php
if (module_exists('entity_translation') && entity_translation_enabled($entity_type, $entity)) {
  if ($handler = entity_translation_get_handler($entity_type, $entity)) {
    $langcodes = array_keys($handler->getTranslations()->data);
  }
}

Reply:
...oh, plus `field_is_translatable()` to check the specific field translatability.

Reply:
Totally 👍  And they can always come back here if they're interested ;-)

Reply:
Done the little helper that @stevetweeddale wanted.